SEED-platform/seed
# feat: allow usage of derived columns in data quality checks

#### Any background context you want to provide?
SEED is adding a feature for users to define columns calculated based on other column values (derived columns)

#### What's this PR do?
Adds the ability for users to use derived columns in data quality rules.
Technically:
- adds `for_derived_column` field to the `Rule` model. I considered just using `table_name` to indicate the rule was for a derived column, but it turned out to overcomplicate things.


#### How should this be manually tested?
1. create a derived column, make sure an inventory has data for the necessary columns so the calculation can be made
2. create a dq rule which references that column
3. run a data quality check from the inventory list page

Also verify creating/editing/deleting dq rules still works for non-derived columns

#### What are the relevant tickets?
#2711 

#### Screenshots (if appropriate)
![Screen Shot 2021-06-15 at 5 47 48 PM](https://user-images.githubusercontent.com/18518728/122137825-c4168a00-ce02-11eb-8f7b-c52b39f3b486.png)
![Screen Shot 2021-06-15 at 5 48 30 PM](https://user-images.githubusercontent.com/18518728/122137833-c678e400-ce02-11eb-838a-881862652630.png)
